code/cfile/cfilearchive.cpp

```diff
@@ -54,18 +54,42 @@ void cf_init_lowlevel_read_code( CFILE * cfile, size_t lib_offset, size_t size,
 	}
 }
 
+//This function checks the file header to see if the file is compressed, if so it fills the correct compression info.
+void cf_check_compression(CFILE* cfile)
+{
+	if (cfile->size <= 16)
+		return;
+	int header=cfread_int(cfile);
+	cfseek(cfile, 0, SEEK_SET);
+	if (comp_check_header(header) == COMP_HEADER_MATCH)
+		comp_create_ci(cfile, header);
+}
+
+//This function is called to cleanup compression info when the cfile handle is reused.
+void cf_clear_compression_info(CFILE* cfile)
+{
+	free(cfile->compression_info.offsets);
+	free(cfile->compression_info.decoderBuffer);
```

Comment:
Ok, added a if to check the header, if is not zero then it means the pointers arent nullptr, if the header is zero then it does nothing as nothing needs to be done.

Reply:
What happens if opening a file fails after the header is detected? Does FSO always quit in such a situation (i.e. due to an `Assertion()`) or is it possible that a CFile block is re-used that has a header but doesn't have an `offsets` and/or `decoderBuffer` pointer?

Reply:
Is not possible to have nullprts along with a header id, after the header is identified you either load the file in the right way or you will trigger some assertion along the way. The two assertions at lz41_load_offsets() makes it almost impossible to pass that point with bad data, and if for some reason you do, the decoder function is going to fail and trigger a file decompression error assertion at cfread,

Reply:
Fair enough I guess. I don't like that all of this relies on the fact that the functions will always be called in the right order since the implementation and inner workings of CFile might change in the future and someone might accidentally break this. However, it seems pretty unlikely for now.